ERCS/erc-6538.md

```diff
@@ -27,65 +27,296 @@ This contract defines an `ERC5564Registry` that stores the stealth meta-address
+    }
+  }
+
+  /// @notice Checks if a signature is valid for a given signer and data hash. If the signer is a
+  /// smart contract, the signature is validated against that smart contract using ERC1271,
+  /// otherwise it's validated using `ECDSA.recover`.
+  /// @dev The following code is from OpenZeppelin's `SignatureChecker.sol` file. Permalink:
+  /// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e70a0118ef10773457f670671baefad2c5ea610d/contracts/utils/cryptography/SignatureChecker.sol
+  function isValidSignatureNow(address signer, bytes32 hash, bytes memory signature)
+    internal
+    view
+    returns (bool)
+  {
+    (address recovered, RecoverError error,) = tryRecover(hash, signature);
+    return (error == RecoverError.NoError && recovered == signer)
+      || isValidERC1271SignatureNow(signer, hash, signature);
   }
 
-  /// @notice Sets the `registrant`s stealth meta-address for the given scheme.
-  /// @param registrant Recipient identifier, such as an ENS name.
-  /// @param scheme An integer identifier for the stealth address scheme.
+  /// @notice Checks if a signature is valid for a given signer and data hash. The signature is
+  /// validated against the signer smart contract using ERC1271.
+  /// @dev The following code is from OpenZeppelin's `SignatureChecker.sol` file. Permalink:
+  /// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e70a0118ef10773457f670671baefad2c5ea610d/contracts/utils/cryptography/SignatureChecker.sol
+  function isValidERC1271SignatureNow(address signer, bytes32 hash, bytes memory signature)
+    internal
+    view
+    returns (bool)
+  {
+    (bool success, bytes memory result) =
+      signer.staticcall(abi.encodeCall(IERC1271.isValidSignature, (hash, signature)));
+    return (
+      success && result.length >= 32
+        && abi.decode(result, (bytes32)) == bytes32(IERC1271.isValidSignature.selector)
+    );
+  }
+
+  /// @dev The following code is from OpenZeppelin's `ECDSA.sol` file. Permalink:
+  /// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e70a0118ef10773457f670671baefad2c5ea610d/contracts/utils/cryptography/ECDSA.sol
+  function tryRecover(bytes32 hash, bytes memory signature)
+    internal
+    pure
+    returns (address, RecoverError, bytes32)
+  {
+    if (signature.length == 65) {
+      bytes32 r;
+      bytes32 s;
+      uint8 v;
+      // ecrecover takes the signature parameters, and the only way to get them
+      // currently is to use assembly.
+      /// @solidity memory-safe-assembly
+      assembly {
+        r := mload(add(signature, 0x20))
+        s := mload(add(signature, 0x40))
+        v := byte(0, mload(add(signature, 0x60)))
+      }
+      return tryRecover(hash, v, r, s);
+    } else {
+      return (address(0), RecoverError.InvalidSignatureLength, bytes32(signature.length));
+    }
+  }
+
+  /// @notice Overload of {ECDSA-tryRecover} that receives the `v`, `r` and `s` signature fields
```

Comment:
Similar comment to the other PR about fixing all natspec to not use OZ's custom style

Reply:
I'll update this once we merge https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5